tests/conftest.py

```diff
@@ -0,0 +1,133 @@
+import argparse
+from dataclasses import dataclass
+from functools import cached_property
+from importlib import import_module
+from sys import version_info as _version_info
+from types import ModuleType
+from typing import List, Type, Union
+
+import pytest
+
+C_EXT_MARK = pytest.mark.c_extension
+PY_38_AND_BELOW = _version_info < (3, 9)
+
+
+@dataclass(frozen=True)
+class PropcacheImplementation:
+    """A facade for accessing importable propcache module variants.
+
+    An instance essentially represents a c-extension or a pure-python module.
+    The actual underlying module is accessed dynamically through a property and
+    is cached.
+
+    It also has a text tag depending on what variant it is, and a string
+    representation suitable for use in Pytest's test IDs via parametrization.
+    """
+
+    is_pure_python: bool
+    """A flag showing whether this is a pure-python module or a C-extension."""
+
+    @cached_property
+    def tag(self) -> str:
+        """Return a text representation of the pure-python attribute."""
+        return "pure-python" if self.is_pure_python else "c-extension"
+
+    @cached_property
+    def imported_module(self) -> ModuleType:
+        """Return a loaded importable containing a propcache variant."""
+        importable_module = "_helpers_py" if self.is_pure_python else "_helpers_c"
+        return import_module(f"propcache.{importable_module}")
+
+    def __str__(self):
+        """Render the implementation facade instance as a string."""
+        return f"{self.tag}-module"
+
+
+@pytest.fixture(
+    scope="session",
+    params=(
+        pytest.param(
+            PropcacheImplementation(is_pure_python=False),
+            marks=C_EXT_MARK,
+        ),
+        PropcacheImplementation(is_pure_python=True),
+    ),
+    ids=str,
+)
+def propcache_implementation(request: pytest.FixtureRequest) -> PropcacheImplementation:
+    """Return a propcache variant facade."""
+    return request.param
+
+
+@pytest.fixture(scope="session")
+def propcache_module(
+    propcache_implementation: PropcacheImplementation,
+) -> ModuleType:
+    """Return a pre-imported module containing a propcache variant."""
+    return propcache_implementation.imported_module
+
+
+def pytest_addoption(
+    parser: pytest.Parser,
+    pluginmanager: pytest.PytestPluginManager,
+) -> None:
+    """Define a new ``--c-extensions`` flag.
+
+    This lets the callers deselect tests executed against the C-extension
+    version of the ``propcache`` implementation.
+    """
+    del pluginmanager
+
+    action: Union[str, Type[argparse.Action]]
```

Comment:
could you use a better var name than an action?

Reply:
0349148ac41673b33b8b9dc722f4791d53ca6439